packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,197 @@
+    }
+
+    function setBond(uint256 _bond) public onlyOwner {
+        // value of the bond required for proposals (must be greater than optimistic oracle final fee)
+        bond = _bond;
+    }
+
+    function setCollateral(IERC20 _collateral) public onlyOwner {
+        // ERC20 token to be used as collateral (must be approved by UMA Store contract).
+        collateral = _collateral;
+    }
+
+    function setRules(string memory _rules) public onlyOwner {
+        // set reference to the rules for the avatar (e.g. an IPFS hash or URI)
+        rules = _rules;
+    }
+
+    function setLiveness(uint256 _liveness) public onlyOwner {
+        // set liveness for disputing proposed transactions
+        liveness = _liveness;
+    }
+
+    function proposeTransactions(Transaction[] memory transactions, bytes memory ancillaryData) public {
+        // create a proposal with a bundle of transactions
+        // note: based in part on the UMA Governor contract
+        // https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/implementation/Governor.sol
+        // note: optional ancillaryData explains the intent of the transactions to make comprehension easier
+        uint256 id = proposals.length;
+        uint256 time = block.timestamp;
+        address proposer = msg.sender;
+
+        // Add a zero-initialized element to the proposals array.
+        proposals.push();
+
+        // Initialize the new proposal.
+        Proposal storage proposal = proposals[id];
+        proposal.requestTime = time;
+        proposal.ancillaryData = ancillaryData;
+
+        // Initialize the transaction array.
+        for (uint256 i = 0; i < transactions.length; i++) {
+            require(transactions[i].to != address(0), "The `to` address cannot be 0x0");
+            // If the transaction has any data with it the recipient must be a contract, not an EOA.
+            if (transactions[i].data.length > 0) {
+                require(_isContract(transactions[i].to), "EOA can't accept tx with data");
+            }
+            proposal.transactions.push(transactions[i]);
+        }
+
+        // Propose a set of transactions to the OO. If not disputed, they can be executed with executeProposal().
+        // docs: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/interfaces/SkinnyOptimisticOracleInterface.sol
+        optimisticOracle.requestAndProposePriceFor(
+            identifier,
+            uint32(time),
+            ancillaryData,
+            collateral,
+            0,
+            bond,
+            liveness,
+            proposer,
+            // Canonical value representing "True"; i.e. the transactions are valid.
+            int256(1e18)
+        );
+
+        emit TransactionsProposed(id, proposer, time);
+    }
+
+    function executeProposal(
+        uint256 _proposalId,
+        uint256 _transactionIndex,
+        Enum.Operation _operation
```

Comment:
It seems that we're letting the caller pick whether this should be a normal call or a delegateCall? Shouldn't this be part of the proposal? I would think that that distinction could make a big difference in the outcome of a transaction.

Reply:
Good point, this should be part of the proposal rather than the execution.

Reply:
Updated this to add operation to the Transaction struct so you can set the operation per transaction.